### PR TITLE
Update error messages with translatable messages and clean up comments

### LINKED ISSUE #23856: Create proper translatable messages for SmallRyeMetricsCDI (mpMetrics-5.0)
The `SmallRyeMetricsCDI` class has multiple opportunities for failure during initialization of SmallRye Metrics and Micrometer related classes (either packaged or user provided via libraryRef attribute).

Currently, from development state, there are only debug statements and a temporary generic (non-translated) error message available to indicate the root problem. There needs to be appropriate messages (errors + info) which indicate the state and root cause of exceptions/failures encountered during startup.

### PR DESCRIPTION
**Pull request by Channyboy**
#build
Fixes #23856

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 1]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+missingSmallRyeClasses.error.CWMMC0012E.explanation=A class or classes from the packaged SmallRye Metrics JAR from the <wlp_home>/lib directory could not be loaded or is missing.
+missingSmallRyeClasses.error.CWMMC0012E.useraction=Review the FFDC logs and exception text to identify the missing class and verify that it exists in the <wlp_home>/lib/io.openliberty.io.smallrye.metrics JAR.
+
+#Configured Micrometer Library is missing dependencies
+missingDependencyClasses.error.CWMMC0013E=CWMMC0013E: MicroProfile Metrics was unable to initialize. A required class for a user provided Micrometer Library is missing.
+missingDependencyClasses.error.CWMMC0013E.explanation=The MicroProfile Metrics feature is loading Micrometer libraries that are defined through the libraryRef attribute. This must also contain all dependencies.
+missingDependencyClasses.error.CWMMC0013E.useraction=Provide the necessary dependencies with the Micrometer Libraries.
+
+#MicroProfile Metrics 5.0 has been configured to use a libraryRef attribute
+libraryRefConfigured.info.CWMMC0014I=CWMMC0014I: MicroProfile Metrics feature has been configured to use the following libraryRef: [{0}]
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
libraryRefConfigured.info.CWMMC0014I=CWMMC0014I: The MicroProfile Metrics feature is configured to use the following libraryRef: [{0}]
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 2]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+libraryRefConfigured.info.CWMMC0014I.explanation=MicroProfile Metrics feature has been configured to use a libraryRef. This referenced library must contain the Micrometer core library, target Micrometer libraries and dependencies.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
libraryRefConfigured.info.CWMMC0014I.explanation=The MicroProfile Metrics feature is configured to use a libraryRef. This referenced library must contain the Micrometer core library, target Micrometer libraries, and dependencies.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 3]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+libraryRefConfigured.info.CWMMC0014I.useraction=No action required.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
libraryRefConfigured.info.CWMMC0014I.useraction=No action is required.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 4]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file may have been deleted or renamed.
+fileNotFound.error.CWMMC0010E.useraction=The server may be corrupt if the JAR file is not present in the <wlp_home>/lib directory. Re-deploy the application server installation.
+
+#Bundle add-on class loader was null
+nullBundleAddOnClassLoader.error.CWMMC0011E=CWMMC0011E: MicroProfile Metrics was unable to initialize. The MicroProfile Metrics runtime was unable to create a required class loader.
+nullBundleAddOnClassLoader.error.CWMMC0011E.explanation=An internal error occured while creating a required class loader.
+nullBundleAddOnClassLoader.error.CWMMC0011E.useraction=Review the FFDC logs and exception text to identify the problem.
+
+#SmallRye Metrics classes not found when loading through reflection
+missingSmallRyeClasses.error.CWMMC0012E=CWMMC0012E: MicroProfile Metrics was unable to initialize. A class from SmallRye Metrics could not be loaded.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
missingSmallRyeClasses.error.CWMMC0012E=CWMMC0012E: The MicroProfile Metrics feature was unable to initialize. A class from SmallRye Metrics was not loaded.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 5]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+missingSmallRyeClasses.error.CWMMC0012E.explanation=A class or classes from the packaged SmallRye Metrics JAR from the <wlp_home>/lib directory could not be loaded or is missing.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
missingSmallRyeClasses.error.CWMMC0012E.explanation=A class or classes from the packaged SmallRye Metrics JAR from the <wlp_home>/lib directory was not loaded or is missing.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 6]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
 disabled.info.CWMMC0009I.explanation=The mp.metrics.prometheus.enabled MicroProfile Config property is set to false.
 disabled.info.CWMMC0009I.useraction=If the disabling of the Prometheus Registry is not intentional, remove the configuration of mp.metrics.prometheus.enabled MicroProfile Config property or explicitly set the property to true.
 
+#Embedded SmallRye Metrics or Micrometer JAR not found in the lib directory
+fileNotFound.error.CWMMC0010E=CWMMC0010E: MicroProfile Metrics was unable to initialize. The required JAR file {0} was not found in the <wlp_home>/lib directory.
+fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file may have been deleted or renamed.
+fileNotFound.error.CWMMC0010E.useraction=The server may be corrupt if the JAR file is not present in the <wlp_home>/lib directory. Re-deploy the application server installation.
+
+#Bundle add-on class loader was null
+nullBundleAddOnClassLoader.error.CWMMC0011E=CWMMC0011E: MicroProfile Metrics was unable to initialize. The MicroProfile Metrics runtime was unable to create a required class loader.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
nullBundleAddOnClassLoader.error.CWMMC0011E=CWMMC0011E: The MicroProfile Metrics were unable to initialize. The MicroProfile Metrics runtime was unable to create a required class loader.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 7]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+nullBundleAddOnClassLoader.error.CWMMC0011E.explanation=An internal error occured while creating a required class loader.
+nullBundleAddOnClassLoader.error.CWMMC0011E.useraction=Review the FFDC logs and exception text to identify the problem.
+
+#SmallRye Metrics classes not found when loading through reflection
+missingSmallRyeClasses.error.CWMMC0012E=CWMMC0012E: MicroProfile Metrics was unable to initialize. A class from SmallRye Metrics could not be loaded.
+missingSmallRyeClasses.error.CWMMC0012E.explanation=A class or classes from the packaged SmallRye Metrics JAR from the <wlp_home>/lib directory could not be loaded or is missing.
+missingSmallRyeClasses.error.CWMMC0012E.useraction=Review the FFDC logs and exception text to identify the missing class and verify that it exists in the <wlp_home>/lib/io.openliberty.io.smallrye.metrics JAR.
+
+#Configured Micrometer Library is missing dependencies
+missingDependencyClasses.error.CWMMC0013E=CWMMC0013E: MicroProfile Metrics was unable to initialize. A required class for a user provided Micrometer Library is missing.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
missingDependencyClasses.error.CWMMC0013E=CWMMC0013E: The MicroProfile Metrics feature was unable to initialize. A class that is required for a user-provided Micrometer Library is missing.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 8]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+missingDependencyClasses.error.CWMMC0013E.explanation=The MicroProfile Metrics feature is loading Micrometer libraries that are defined through the libraryRef attribute. This must also contain all dependencies.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
missingDependencyClasses.error.CWMMC0013E.explanation=The MicroProfile Metrics feature is loading Micrometer libraries that are defined through the libraryRef attribute that is containing all the dependencies.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 9]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
 disabled.info.CWMMC0009I.explanation=The mp.metrics.prometheus.enabled MicroProfile Config property is set to false.
 disabled.info.CWMMC0009I.useraction=If the disabling of the Prometheus Registry is not intentional, remove the configuration of mp.metrics.prometheus.enabled MicroProfile Config property or explicitly set the property to true.
 
+#Embedded SmallRye Metrics or Micrometer JAR not found in the lib directory
+fileNotFound.error.CWMMC0010E=CWMMC0010E: MicroProfile Metrics was unable to initialize. The required JAR file {0} was not found in the <wlp_home>/lib directory.
+fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file may have been deleted or renamed.
+fileNotFound.error.CWMMC0010E.useraction=The server may be corrupt if the JAR file is not present in the <wlp_home>/lib directory. Re-deploy the application server installation.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
fileNotFound.error.CWMMC0010E.useraction=The server might be corrupted if the JAR file is not present in the <wlp_home>/lib directory. Redeploy the application server installation.
```

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 10]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
 disabled.info.CWMMC0009I.explanation=The mp.metrics.prometheus.enabled MicroProfile Config property is set to false.
 disabled.info.CWMMC0009I.useraction=If the disabling of the Prometheus Registry is not intentional, remove the configuration of mp.metrics.prometheus.enabled MicroProfile Config property or explicitly set the property to true.
 
+#Embedded SmallRye Metrics or Micrometer JAR not found in the lib directory
+fileNotFound.error.CWMMC0010E=CWMMC0010E: MicroProfile Metrics was unable to initialize. The required JAR file {0} was not found in the <wlp_home>/lib directory.
+fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file may have been deleted or renamed.
```
Comment (ramkumar-k-9286) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]:
```suggestion
fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file might be deleted or renamed.
```

**Review by fmhwong — COMMENTED** [review]
I noticed some messages use "MicroProfile Metrics" and some use "MicroProfile Metrics feature".  Is "MicroProfile Metrics" same as "MicroProfile Metrics feature" in the messages?

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 9 above)
Comment (donbourne) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: The xxx JAR file, normally found in the <wlp_home>/lib directory, is not present. Re-deploy the application server installation.

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 10 above)
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: could we include the name of the expected jar file in this message?

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 4 above)
Comment (donbourne) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: MicroProfile Metrics feature was unable...

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 7 above)
Comment (donbourne) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: The MicroProfile Metrics feature was ...

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 11]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
 disabled.info.CWMMC0009I.explanation=The mp.metrics.prometheus.enabled MicroProfile Config property is set to false.
 disabled.info.CWMMC0009I.useraction=If the disabling of the Prometheus Registry is not intentional, remove the configuration of mp.metrics.prometheus.enabled MicroProfile Config property or explicitly set the property to true.
 
+#Embedded SmallRye Metrics or Micrometer JAR not found in the lib directory
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops:77]: don't forget to change the copyright year on this file

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/internal/SmallryeMetricsCDIMetadata.java [hunk 12]
```diff
@@ -79,8 +79,13 @@ public class SmallryeMetricsCDIMetadata implements CDIExtensionMetadata {
 
     /**
      * Used by SharedMetricRegistries to figure out if it should provide any
-     * MetricRegistries. Since we cannot stop activation, this is a flag to use to
-     * "stop" it from emitting dramatic runtime errors.
+     * MetricRegistries. Also used to prevent this Service-Component from providing
+     * the extension class during the call-back mechanism as part of the
+     * {@link CDIExtensionMetadataSince} service. Throwing an exception during
+     * activation will prompt OSGI to re-activate over and over. Depending on the
+     * failure that occurs during activation, this is a flag to used to "stop" any
```
Comment (donbourne) [inline dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/internal/SmallryeMetricsCDIMetadata.java]: "to used to"

File: dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/internal/SmallryeMetricsCDIMetadata.java [hunk 13]
```diff
@@ -143,46 +136,40 @@ protected void activate(ComponentContext context, Map<String, Object> properties
...
-            }
-
-            // TODO: Make actual message
-            Tr.error(tc, "UNABLE TO INITIALIZE MICROPROFILE METRICS 5.0 FEATURE DUE TO: " + e);
-            isSuccesfulActivation = false;
             /*
-             * Trying to throw an exception to stop activation does not work. OSGI will keep
-             * trying to restart service. This will result in multiple console outputs.
-             * throw new Exception("Unable to initialize MicroProfile Metrics 5.0 feature");
+             * Probable cause: Missing dependent class(es) during runtime initialization.
```
Comment (donbourne) [inline dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/internal/SmallryeMetricsCDIMetadata.java:153]: I really like these "Probable cause" comments!

**Review by donbourne — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 8 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: The error is about a missing class (i.e. JAR dependency) inthe directory that is referenced by the `libraryRef` attribute. Not sure if the suggestion directly explains to the customer what is wrong? 
Perhaps if I switch it to `"... and must contain all the dependencies.`

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 4 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: changed to "The MicroProfile Metrics feature was"

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 6 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: changed to "The MicroProfile Metrics feature was"

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 7 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: changed to "The MicroProfile Metrics feature was"

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops (hunk 10 above)
Comment (Channyboy) [reply dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: This message is used when checking for the packaged smallrye metrics JAR and packaged micrometer jar. The {0} is the param used to identify the jar. The parameter would apply to the logged message. The explanation and the user-action would have to simply refer to JAR.

**Comment by Channyboy** [comment]
#build #libby

**Review by fmhwong — DISMISSED** [review]
LGTM

File: dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops [hunk 14]
```diff
@@ -74,6 +74,31 @@ disabled.info.CWMMC0009I=CWMMC0009I: The /metrics endpoint is not available beca
...
+fileNotFound.error.CWMMC0010E=CWMMC0010E: The MicroProfile Metrics feature was unable to initialize. The required JAR file {0} was not found in the <wlp_home>/lib directory.
+fileNotFound.error.CWMMC0010E.explanation=The JAR file was not found in the <wlp_home>/lib directory. The file might be deleted or renamed.
+fileNotFound.error.CWMMC0010E.useraction=The JAR, normaly found in <wlp_home>/lib directory, is not present. Redeploy the application server installation.
```
Comment (donbourne) [inline dev/com.ibm.ws.microprofile.metrics.common/resources/com/ibm/ws/microprofile/metrics/resources/Metrics.nlsprops]: normaly -> normally

**Review by donbourne — CHANGES_REQUESTED** [review]
see minor comment...

**Comment by Channyboy** [comment]
#build #libby

**Review by donbourne — APPROVED** [review]

**Review by ramkumar-k-9286 — APPROVED** [review]

**Pull request merged**